src/esp/sensor/CameraSensor.cpp

```diff
@@ -20,6 +20,7 @@ CameraSensor::CameraSensor(scene::SceneNode& cameraNode,
       baseProjMatrix_(Magnum::Math::IdentityInit),
       zoomMatrix_(Magnum::Math::IdentityInit) {
   setProjectionParameters(spec);
+  renderCamera_ = new gfx::RenderCamera(cameraNode);
```

Comment:
After you created this renderCamera, you have to set it.
You will have to set the projection matrix, view port once since it will not change during the lifetime of the sensor.
By doing so, you do not have to set the projection matrix again and again in the `getObservation()` function.